internal/endorser/endorser_test.go

```diff
@@ -92,13 +99,80 @@ func TestGenerateEndorsement_FailingSingleRemoteProvenanceEndorsement(t *testing
 	}
 
 	provenances := []string{"https://github.com/project-oak/transparent-release/blob/main/testdata/provenance.json"}
-	_, err := GenerateEndorsement(binaryHash, validity, provenances)
+	reference := verifier.ProvenanceIR{
+		BinarySHA256Digests: []string{binaryHash},
+	}
+	_, err := GenerateEndorsement(reference, validity, provenances)
 	want := "could not load provenances"
 	if err == nil || !strings.Contains(err.Error(), want) {
 		t.Fatalf("got %q, want error message containing %q,", err, want)
 	}
 }
 
+func TestGenerateEndorsement_MultipleNotVerifiedOrConsistent(t *testing.T) {
+	tomorrow := time.Now().AddDate(0, 0, 1)
+	nextWeek := time.Now().AddDate(0, 0, 7)
+	validity := amber.ClaimValidity{
+		NotBefore: &tomorrow,
+		NotAfter:  &nextWeek,
+	}
+
+	tempPath1, err := copyToTemp("../../testdata/provenance.json")
+	if err != nil {
+		t.Fatalf("Could not load provenance: %v", err)
+	}
+	tempPath1Variant, err := copyToTemp("../../testdata/provenance_variant.json")
+	if err != nil {
+		t.Fatalf("Could not load provenance: %v", err)
+	}
+	tempPath2, err := copyToTemp("../../testdata/different_provenance.json")
+	if err != nil {
+		t.Fatalf("Could not load provenance: %v", err)
+	}
+	tests := []struct {
+		reference       []string
+		provenancePaths []string
+		want            string
+	}{
+		// Provenances do not contain reference binary SHA256 digest, but are consistent.
+		{
+			reference:       []string{binaryHash + "_diff"},
+			provenancePaths: []string{"file://" + tempPath1, "file://" + tempPath1Variant},
+			want:            "do not contain the actual binary SHA256 digest",
+		},
+		// Provenances contain reference binary SHA256 digest, but are inconsistent.
```

Comment:
Don't we need a scenario where there are multiple consistent provenances, and multiple reference values? 
That is the most realistic scenario. 
The two interesting scenarios are (1) where provenances are consistent, and the digest is contained in the reference values, and (2) the provenances are consistent, but the digest is not contained in the reference values.

Reply:
I think (1) we had already (`TestGenerateEndorsement_MultipleValidEndorsement`) and (2) is the first "// Provenances do not contain reference binary SHA256 digest, but are consistent."

Reply:
(1) is different from `TestGenerateEndorsement_MultipleValidEndorsement` in that in `TestGenerateEndorsement_MultipleValidEndorsement` we have only a single reference value; I am looking for a scenario with multiple reference values for the digest.

(2) is more generic than `"// Provenances do not contain reference binary SHA256 digest, but are consistent."`, in that (2) has multiple reference values. But actually, I was a bit confused by the wording of that comment. The current wording sounds like that the "provenances do not contain a SHA256 digest", which is a different negative test.